public/app/features/dashboard-scene/scene/layout-responsive-grid/ResponsiveGridLayoutManager.test.ts

```diff
@@ -0,0 +1,53 @@
+import { SceneCSSGridLayout, SceneQueryRunner, VizPanel } from '@grafana/scenes';
+
+import { findVizPanelByKey } from '../../utils/utils';
+import { DashboardScene } from '../DashboardScene';
+
+import { ResponsiveGridItem } from './ResponsiveGridItem';
+import { ResponsiveGridLayoutManager } from './ResponsiveGridLayoutManager';
+
+describe('ResponsiveGridLayoutManager', () => {
+  it('Should clone the layout', () => {
+    const { manager } = setup();
+    const clone = manager.cloneLayout('foo', true) as ResponsiveGridLayoutManager;
+
+    expect(clone).not.toBe(manager);
+    expect(clone.state.layout).not.toBe(manager.state.layout);
+    expect(clone.state.layout.state.children).not.toBe(manager.state.layout.state.children);
+    expect(clone.state.layout.state.children.length).toBe(manager.state.layout.state.children.length);
+
+    const panelA = findVizPanelByKey(clone, 'foo/grid-item-0/panel-0');
```

Comment:
We should preserve the panel IDs (check the suggestion below).

```suggestion
    const panelA = findVizPanelByKey(clone, 'foo/grid-item-0/panel-1');
```